src/main/ruby/core/hash.rb

```diff
@@ -94,6 +94,7 @@ def self._constructor_fallback(*args)
 
     hash
   end
+  private_class_method :_constructor_fallback
```

Comment:
This is an improvement, but I think it should just move out to `hash_operations.rb`.

Reply:
Out of scope?
I tried to make `Hash.[]` a primitive but it wasn't practical.

Reply:
Yeah. We can clean it up after.